### PR TITLE
Glowshrooms now have colored light, tweaks glowshroom/glowberry lighting range

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
rscadd: Glowshrooms now have colored light.
tweak: Tweaked the potency scaling for glowshroom/glowberry light; high-potency plantss no longer light up a huge area, but are slightly brighter.
/:cl:

Fixed a bug where the first glowshroom planted wouldn't have light properly based on potency.

Max potency shrooms

This is actually less wide-ranging than it used to be; light_range at 100 potency from 10 to 4, light_power from 1 to 4

### CODE REVIEW

**Comment by GunHog** [comment]
You should include the part where you are nerfing them in your title.

**Comment by PKPenguin321** [comment]
Looks super fucking cool, I'm down for nerfing them for this aesthetic

**Comment by GunHog** [comment]
Eh, I guess. I admit that they do look RAD. (As in radioactive)

**Comment by Dorsisdwarf** [comment]
there. 


👍

**Comment by Supermichael777** [comment]
are the red ones RED?

**Comment by bobdobbington** [comment]
Can you please split this PR into one for the nerf and one for the new lighting? The new lighting is very good and the 80% range nerf is way too much. I understand that now that the light is colored it'll be more jarring to have wide ranges but there's gotta be a better option than 2 at max potency. I'd just make glowberries instead.

**Comment by ChangelingRain** [comment]
It's actually 2.5 whoops
But uh, 10 is super hilarious because one shroom lights up the entire screen and then some, like, I can get you a picture of that

**Comment by ChangelingRain** [comment]
Also do account for the fact that I made the actual glowshroom item glow as well, so it works as basically a flashlight in addition to lighting.

**Comment by bobdobbington** [comment]
Glowshrooms already glowed as a result of their having the Bioluminescent trait, so that's not a buff. I agree that 10 doesn't make sense with colored lights but even going from a radius 10 down to a radius 5 range would mean going from ~300 potentially covered tiles down to more like ~80. Dropping it to 2.5 drops the covered area down to ~20 tiles, giving these less than a tenth the effective range of pre-nerf levels.

File: code/game/objects/effects/glowshroom.dm [hunk 1]
```diff
@@ -35,7 +38,10 @@ var/list/blacklisted_glowshroom_turfs = typecacheof(list(
 
 /obj/structure/glowshroom/New()
 	..()
-	set_light(round(potency/10))
+	addtimer(CALLBACK(src, .proc/setup), 0) //run this on the next tick so we can set vars first
+
+/obj/structure/glowshroom/proc/setup()
```
Comment (Cyberboss) [inline code/game/objects/effects/glowshroom.dm]: Remove the timer and use Initialize

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/glowshroom.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/game/objects/effects/glowshroom.dm]: Doesn't work, smart guy; it can't set the vars before set_light and the addtimer is called.

**Comment by ChangelingRain** [comment]
Tweaked it to be a maximum of 4 range, 4 brightness, which should be more than enough given how many of these you can produce and that they spread somewhat if planted.

**Comment by bobdobbington** [comment]
Alright, I can live with that compromise. Can we bump Strong to 0.06 so it's an exact double of regular again? That's enough for me to be happy with this PR.

**Comment by ChangelingRain** [comment]
Due to how lights scale, increasing the light range by 2 has a major effect on the brightness and increasing it to that would mean it'd be full screen light at max potency. I think that's kind of a bit much.

**Comment by bobdobbington** [comment]
Yeah, the idea is that at max potency, a glowberry should light up the whole screen. That's what they do now. This glowshroom change shouldn't impact the usefulness of pocket glowberries as maint/mining lights.

**Comment by ExcessiveUseOfCobblestone** [comment]
A T O M I Z E 

Also you shouldn't be using a plant as a replacement for the light, you should be using it's ability to spread + light, so I think the balance change not being a full screen is ok.

**Comment by ChangelingRain** [comment]
You can't spread glowberries but they've got huge fucking range compared to basically every other item light in the game even with this one-tile nerf

**Comment by bobdobbington** [comment]
Glowberries aren't glowshrooms. I can see the relevance between changing the color of glowshroom lights and nerfing the range but this glowberry nerf seems just tacked on because you feel that botany should have less powerful lights? It has no relevance whatsoever to the change that is the reason for this PR.

**Comment by bobdobbington** [comment]
Changing the title doesn't make it any more related. The impetus for all this was that you changed the color of glowshrooms, which resulted in them needing to be tweaked. Why are you so insistent on nerfing this? You haven't given a reason beyond "huge fucking range" and "that's kind of a bit much". I want to be able to see the entire screen as plantpod on lavaland with a max potency glowberry. That's not unreasonable at all. Why do you think glowberries are a balance issue, and why are you lumping a glowberry nerf into an unrelated glowshroom PR?

**Comment by XDTM** [comment]
Glowberry genes can be transferred to glowshrooms, i'm pretty sure.

**Comment by bobdobbington** [comment]
Shouldn't impact the spreading glowshrooms on the ground, I believe. At most by genemodding you'd be able to get a glowshroom with equivalent pocket brightness, not spreading glowberries.

**Comment by Cheridan** [comment]
jesus bob lay off

**Pull request merged**